examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+    step: int,
+    margin: int,
+) -> torch.utils.data.DataLoader:
+    """
+    Setup function for a infer data loader.
+
+    Args:
+        model: a ``PunctuationCapitalizationModel`` instance for which data loader is created.
+        queries: lower cased text without punctuation
+        batch_size: batch size to use during inference
+        max_seq_length: length of segments into which queries are split. ``max_seq_length`` includes ``[CLS]`` and
+            ``[SEP]`` so every segment contains at most ``max_seq_length-2`` tokens from input a query.
+        step: number of tokens by which a segment is offset to a previous segment. Parameter ``step`` cannot be greater
+            than ``max_seq_length-2``.
+        margin: number of tokens near the edge of a segment which label probabilities are not used in final prediction
+            computation.
+    Returns:
+        A pytorch DataLoader.
+    """
+    if max_seq_length is None:
+        max_seq_length = model._cfg.dataset.max_seq_length
+    if step is None:
+        step = model._cfg.dataset.step
+    if margin is None:
+        margin = model._cfg.dataset.margin
+
+    dataset = BertPunctuationCapitalizationInferLongDataset(
+        tokenizer=model.tokenizer, queries=queries, max_seq_length=max_seq_length, step=step, margin=margin
+    )
+    return torch.utils.data.DataLoader(
+        dataset=dataset,
+        collate_fn=dataset.collate_fn,
+        batch_size=batch_size,
+        shuffle=False,
+        num_workers=model._cfg.dataset.num_workers,
+        pin_memory=model._cfg.dataset.pin_memory,
+        drop_last=False,
+    )
+
+
+def _move_acc_probs_to_token_preds(
+    pred: List[int], acc_prob: np.ndarray, number_of_probs_to_move: int
+) -> Tuple[List[int], np.ndarray]:
+    """
+    ``number_of_probs_to_move`` rows in the beginning are removed from ``acc_prob``. From every remove row the label
+    with the largest probability is selected and appended to ``pred``.
+    Args:
+        pred: list with ready label indices for a query
+        acc_prob: numpy array of shape ``[number_of_words_for_which_probabilities_are_accumulated, number_of_labels]``
+        number_of_probs_to_move: int
+    Returns:
+        pred: list with ready label indices for a query
+        acc_prob: numpy array of shape
+            ``[number_of_words_for_which_probabilities_are_accumulated - number_of_probs_to_move, number_of_labels]``
+    """
+    if number_of_probs_to_move > acc_prob.shape[0]:
+        raise ValueError(
+            f"Not enough accumulated probabilities. Number_of_probs_to_move={number_of_probs_to_move} "
+            f"acc_prob.shape={acc_prob.shape}"
+        )
+    if number_of_probs_to_move > 0:
+        pred = pred + list(np.argmax(acc_prob[:number_of_probs_to_move], axis=-1))
+    acc_prob = acc_prob[number_of_probs_to_move:]
+    return pred, acc_prob
+
+
+def _update_accumulated_probabilities(acc_prob: np.ndarray, update: np.ndarray) -> np.ndarray:
```

Comment:
move to the model class or create a utils file inside https://github.com/NVIDIA/NeMo/tree/main/nemo/collections/nlp/models/token_classification